### PR TITLE
add flush to the stream at the time of dispose

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

File: Ds3/Helpers/Streams/DisposableStream.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        protected override void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _stream.Flush();
```
Comment (rpmoore) [inline Ds3/Helpers/Streams/DisposableStream.cs]: You need to pass true into the flush to do what we want.

File: Ds3/Helpers/Streams/DisposableStream.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+using System;
+using System.IO;
+
+namespace Ds3.Helpers.Streams
+{
+    internal class DisposableStream : Stream, IDisposable
+    {
+        private readonly Stream _stream;
```
Comment (rpmoore) [inline Ds3/Helpers/Streams/DisposableStream.cs]: This needs to be a FileStream to get the right flushing behavior that we want.

**Comment by rpmoore** [comment]
We should also change https://github.com/SpectraLogic/ds3_net_sdk/blob/master/Ds3/Helpers/FileHelpers.cs#L51 and https://github.com/SpectraLogic/ds3_net_sdk/blob/master/Ds3/Helpers/FileHelpers.cs#L92 so that they return the FileStream wrapped in the new DisposableStream.

**Comment by shabtaisharon** [comment]
weird since I change those...

**Comment by shabtaisharon** [comment]
there are 3 files changed in the pull request including what you said

File: Ds3/Helpers/Streams/DisposableStream.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+using System;
+using System.IO;
+
+namespace Ds3.Helpers.Streams
+{
+    internal class DisposableStream : Stream, IDisposable
```
Comment (rpmoore) [inline Ds3/Helpers/Streams/DisposableStream.cs]: Since this is specific to FileStreams we may want to call it DisposableFileStream instead

**Pull request merged**